src/main/java/org/testng/xml/XmlTest.java

```diff
@@ -388,7 +389,12 @@ public String getParameter(String name) {
   }
 
   public void setParallel(XmlSuite.ParallelMode parallel) {
-    m_parallel = parallel;
+    if (parallel == XmlSuite.ParallelMode.TRUE || parallel == XmlSuite.ParallelMode.FALSE) {
```

Comment:
Looks like both `XmlTest` and `XmlSuite` could share this code. Maybe extract it in a common `parseParallelValue`?